assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js

```diff
@@ -35,22 +35,32 @@ const { useSelect } = Data;
 const WidgetContextRenderer = ( props ) => {
 	const { id, slug, className, Header, Footer } = props;
 
-	const widgetAreas = useSelect( ( select ) => {
-		if ( slug ) {
-			return select( CORE_WIDGETS ).getWidgetAreas( slug );
+	const { widgetAreas, isActive } = useSelect( ( select ) => {
```

Comment:
@eugene-manuilov, thanks for letting me know about this convention. I'll switch to using separate hooks in this PR and future code, although I do have a couple of questions/points about it.

Firstly, I think we could improve this convention a bit with a helper hook, something like this would cut down the repetition a bit and be a bit easier on the eyes, IMHO:

```js
const widgetArea = useSelectFrom( CORE_WIDGETS ).getWidgetArea( slug );
```

Also, this point didn't come up during the review here where I did use a `useSelect` with multiple calls to `select` - https://github.com/google/site-kit-wp/pull/4294/files#diff-ce00642e223346f597c625a87b208e9f359fe7c4b2f85f61970b863e5a20088a - and when discussing this custom hook approach on a call with @aaemnnosttv he suggested we might want to adopt this approach as a new convention, so maybe there's a bit of further discussion to be had around this...

Reply:
Done :+1:

Reply:
> Firstly, I think we could improve this convention a bit with a helper hook, something like this would cut down the repetition a bit and be a bit easier on the eyes...

Yes, I agree with you that the current formatting isn't perfect and I have also tried to change it a few times in my PRs but currently this is how we do it and I don't think we are going to change it in the nearest future because we have more higher priority tasks. Nevertheless I like your idea and encourage you to create a ticket for it. Once we have less work to do, we can get back to it and perhaps implement it at some point in the future.

> Also, this point didn't come up during the review here where I did use a useSelect with multiple calls to select...

Yes, we have some places in our codebase where we use combined select hooks, but those are more of an exception than a rule. If you still think strongly about it, feel free to raise this question on our engineering meeting call and we can discuss it there.

Reply:
Thanks @eugene-manuilov. I've created an issue for the helper hook proposal, https://github.com/google/site-kit-wp/issues/4378, and will also bring the `useSelect` topic up in one of the engineering meetings to discuss :+1: